UnitTests/CommonTestUtils/ConfigureJoinableTaskFactoryAttribute.cs

```diff
@@ -72,16 +85,29 @@ public void AfterTest(ITest test)
 
                 _denyExecutionSynchronizationContext?.ThrowIfSwitchOccurred();
             }
+            catch (Exception ex) when (_threadException != null)
+            {
+                // ignore the follow-up exception
+                LoggingService.Log($"{nameof(AfterTest)} ignoring", ex);
+            }
             finally
             {
+                AppSettings.CheckForUpdates = _checkForUpdatesOriginalValue;
                 Application.ThreadException -= HandleApplicationThreadException;
-                _threadException?.Throw();
+                if (_threadException != null)
+                {
+                    LoggingService.Log($"{nameof(AfterTest)} rethrow", _threadException.SourceException);
+                    _threadException.Throw();
+                }
```

Comment:
💡 The original code was fine here. The test log will already report the exceptions.

Reply:
This is a remainder of the debug instrumentation. I am undecided whether we should keep all the debug instrumentation or not.